app.py

```
@@ -3675,6 +3296,113 @@ def _fetch_relations(graph: Any, memory_id: str) -> List[Dict[str, Any]]:
+            bfs: true,
+            filterStartNode: true
+        }}) YIELD path
+        WITH DISTINCT related
+        RETURN related
+        ORDER BY coalesce(related.importance, 0.0) DESC, coalesce(related.timestamp, '') DESC
+        LIMIT $limit
+    """
+
+    # If APOC is unavailable, fall back to simple 1..depth traversal without apoc
+    fallback_query = f"""
+        MATCH (m:Memory {{id: $id}}){'-[r' + rel_pattern + f'*1..$max_depth]-' if rel_pattern else '-[r*1..$max_depth]-'}(related:Memory)
```

Ruff (0.14.3)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)